File: tests/kotlin/lambdas/test_higher_order_function.kt
// vybe-test: kotlin/lambdas/test_higher_order_function
// origin: languages/kotlin/tests/kotlin/test_lambdas.rs

fun execute(a: Int, b: Int, op: (Int, Int) -> Int): Int {
            return op(a, b)
        }

        fun __check(got: String, want: String) {
    if (got != want) {
        println("FAIL: want [" + want + "] got [" + got + "]")
        throw Exception("assertion failed")
    }
}

fun main() {
            val sum = execute(15, 25, { x, y -> x + y })
            __check((sum).toString(), "40")
        }

File: tests/js/features/test_higher_order_function.js
// vybe-test: js/features/test_higher_order_function
// origin: languages/js/tests/js/js_features_test.rs

function __line(...args) {
    // console.log joins its arguments with a single space. String() is the
    // coercion Vybe's logging host applies to each one.
    return args.map(String).join(" ");
}

function __check(got, want) {
    if (got !== want) {
        console.log("FAIL: want [" + want + "] got [" + got + "]");
        throw new Error("assertion failed");
    }
}

function apply(f, x) { return f(x); }
        let double = (x) => x * 2;
        __check(__line(apply(double, 21)), "42");
